arch/network_ws.c

```diff
@@ -242,7 +242,10 @@ ServerNetworkLayerWS_start(UA_ServerNetworkLayer *nl, const UA_String *customHos
                          "Could not get the hostname");
         }
     }
-    UA_String_copy(&du, &nl->discoveryUrl);
+    // we need discoveryUrl.data as a null-terminated string for vhost_name
+    nl->discoveryUrl.data = (UA_Byte *)UA_malloc(du.length+1);
+    strcpy((char *)nl->discoveryUrl.data, discoveryUrlBuffer);
```

Comment:
du.length is known.
We should use strncpy here.

Reply:
Codacy doesn't like strncpy:

> Easily used incorrectly; doesn't always \0-terminate or check for invalid pointers [MS-banned] (CWE-120).

Reply:
@jpfr So what should I do? Leave it or revert it?

And what is going on with CI? Again I believe that this PR could not have affected any tests since UA_ENABLE_WEBSOCKET_SERVER is not on when testing.